tests/test_utils/asserts.py

```diff
@@ -19,22 +19,38 @@
 import logging
 import re
 import traceback
+import warnings
 from collections import Counter
 from contextlib import contextmanager
+from typing import TYPE_CHECKING
 
 from sqlalchemy import event
 
 # Long import to not create a copy of the reference, but to refer to one place.
 import airflow.settings
 
+if TYPE_CHECKING:
+    from unittest import TestCase
+
 log = logging.getLogger(__name__)
 
 
-def assert_equal_ignore_multiple_spaces(case, first, second, msg=None):
+def assert_equal_ignore_multiple_spaces(case: TestCase | None, first, second, msg=None):
     def _trim(s):
         return re.sub(r"\s+", " ", s.strip())
 
-    return case.assertEqual(_trim(first), _trim(second), msg)
+    if case:
+        warnings.warn(
+            "Passing `case` has no effect and will be remove in the future. "
+            "Please set to `None` for avoid this warning.",
+            FutureWarning,
+            stacklevel=3,
+        )
+
+    if not msg:
+        assert _trim(first) == _trim(second)
+    else:
+        assert _trim(first) == _trim(second), msg
```

Comment:
Replace usage `unittests.TestCase.assertEqual` by assert.
Because some of tests use this helper (google, amazon, apache) I add warning as a reminder for further migration to `pytest`.